package.json

```diff
@@ -65,6 +65,7 @@
     "@mui/material": "5.11.12",
     "@node-steam/vdf": "2.2.0",
     "@shockpkg/icon-encoder": "2.1.3",
+    "@tanstack/react-query": "^5.71.10",
```

Comment:
do we HAVE to add this new dependency? I'm always worried about adding new dependencies and I see it's used to fetch the list of wines, and we already do that in other places without this dependency, can't we re-use what we did in other places instead?

Reply:
I didn't like to use React Query at first, until I started to doing it. 
Their error handling, loading state for any request we want to and it is also not a heavy dependency. I think we should start using it more, it might improve a lot the way we handle request on the frontend, especially handling errors and refreshes.

Reply:
I second this. Looking at how this dependency is used right now, I don't see anything that *can't* be implemented with regular React hooks or our `GlobalStateV2`.  
I'm willing to compromise on this staying in for now, but in that case I'll look into removing it again in a future PR

Reply:
can we introduce this in a different PR maybe? I think it will help keep this PR simpler (I think it will actually solve your issues with the workflow if the lockfile is not changed?)

I've never used react query, not really against it but I think it's better if it's done in another PR, and I know there are alternatives  and also Suspense

Reply:
they serve diferent porpuses @arielj 
I can remove ofc, but would be something good to have on Heroic since we need to do a lot of requests and the way react-query deals with errors and loading state for all requests is nice.
Normally we would need to use a setState and a useEffect to control the loading state of a request, with react-query we won't need that.